acceptance/repository-cloudant/src/__tests__/cloudant.datasource.ts

```diff
@@ -9,12 +9,12 @@ const connector = require('loopback-connector-cloudant');
 
 export const CLOUDANT_CONFIG: DataSourceOptions = {
   connector,
-  host: process.env.CLOUDANT_HOST || 'localhost',
-  port: process.env.CLOUDANT_PORT || 5984,
-  username: process.env.CLOUDANT_USER || 'admin',
-  password: process.env.CLOUDANT_PASSWORD || 'pass',
-  database: process.env.CLOUDANT_DATABASE || 'repository_tests',
-  url: process.env.CLOUDANT_URL || 'http://localhost:5984',
+  host: process.env.CLOUDANT_HOST ?? 'localhost',
+  port: process.env.CLOUDANT_PORT ?? 5984,
```

Comment:
IIUC, with this change in place, we will convert `CLOUDANT_PORT=0` to `port: 0`. Before this change, we would set `port: 5984`.

Do we care about this change?

The same comment applies to other similar places.

Reply:
I think the new behavior better reflects our intention. `??` is stricter (==null) than `||` (falsy).

Reply:
I understand the difference between `??` and `||`.

What I am trying to say is that port `0` is not a valid value for clients. At the moment, we are replacing this invalid value with the default port 5984, which seems like a good solution to me. With your change in place, we will use the invalid port. To me, that seems like a change for worse.

@strongloop/sq-lb-apex thoughts?

Reply:
IMO, the purpose of using `||` or `??` is not to exclude invalid values. For example, `0`, `1`, `2` are invalid values set explicitly by the client but we cannot tell. It's better to expose such errors to the client instead of hiding them.

The intention here should be:

1. If the env var is set (!=null), we use it as is.
2. Otherwise use 5984.

Reply:
Fair enough 👍